### DISCUSSION TITLE
Progress Listener code for upload

### DISCUSSION THREAD

**Discussion by Conjuror1279** [aws/aws-sdk-kotlin#926] [Q&A]
Hi, This is in reference to the issue https://github.com/awslabs/aws-sdk-kotlin/issues/490. Is there someplace where I can find a code sample for the ProgressListenerBody class, I am not able to understand how to add a progress listener in the above code. The function modifyBeforeTransmit is only called once and doesnot give me the progress about the upload. I've been stuck at this for quite a while, please help me figure this out.

Also, is there a way to receive or generate the presigned url of the uploaded file using this kotlin sdk?

**Comment by ianbotsf** [comment]
Tracking discussion in #490.
